File: app/src/main/java/com/kakeibo/ui/Query.java
//package com.kakeibo.ui;
//
//import android.os.Parcel;
//import android.os.Parcelable;
//
//import java.util.HashMap;
//import java.util.Map;
//
//public class Query implements Parcelable {
//    public static final int QUERY_TYPE_NEW = 0;
//    public static final int QUERY_TYPE_SEARCH = 1;
//
//    private static int type;
//    private static Map<Integer, String> queryCs = new HashMap<>();
//    private static String queryC;
//    private static String queryD;
//
//    public Query (int arg) { type = arg; }
//
//    public int getType() { return type; }
//
//    public Map<Integer, String> getQueryCs() { return queryCs; }
//
//    public String getQueryC() { return queryC; }
//
//    public String getQueryD() { return queryD; }
//
//    public void setQueryCs(Map<Integer, String> queries) { queryCs = queries; }
//
//    public void setQueryC(String query) { queryC = query; }
//
//    public void setQueryD(String query) { queryD = query; }
//
//    /******** parcel ********/
//    private Query (Parcel in) {
//        type = in.readInt();
//        in.readMap(queryCs, String.class.getClassLoader());
//        queryC = in.readString();
//        queryD = in.readString();
//    }
//
//    @Override
//    public int describeContents() {
//        return 0;
//    }
//
//    @Override
//    public void writeToParcel(Parcel dest, int flags) {
//        dest.writeInt(type);
//        dest.writeMap(queryCs);
//        dest.writeString(queryC);
//        dest.writeString(queryD);
//    }
//
//    public static final Parcelable.Creator CREATOR = new Parcelable.Creator() {
//        public Query createFromParcel(Parcel in) {
//            return new Query(in);
//        }
//
//        public Query[] newArray(int size) {
//            return new Query[size];
//        }
//    };
//}

File: app/src/main/java/com/kakeibo/ui/model/Query.kt
package com.kakeibo.ui.model

import android.os.Parcel
import android.os.Parcelable
import java.util.*

class Query : Parcelable {

    constructor(arg: Int) {
        Companion.type = arg
    }

    val type: Int
        get() = Companion.type

    var queryCs: Map<Int?, String?>
        get() = Companion.queryCs
        set(queries) {
            Companion.queryCs = queries
        }
    var queryC: String?
        get() = Companion.queryC
        set(query) {
            Companion.queryC = query
        }
    var queryD: String?
        get() = Companion.queryD
        set(query) {
            Companion.queryD = query
        }

    /******** parcel  */
    private constructor(`in`: Parcel) {
        Companion.type = `in`.readInt()
        `in`.readMap(Companion.queryCs, String::class.java.classLoader)
        Companion.queryC = `in`.readString()
        Companion.queryD = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(Companion.type)
        dest.writeMap(Companion.queryCs)
        dest.writeString(Companion.queryC)
        dest.writeString(Companion.queryD)
    }

    companion object {
        const val QUERY_TYPE_NEW = 0
        const val QUERY_TYPE_SEARCH = 1
        private var type: Int = 0
        private var queryCs: Map<Int?, String?> = HashMap()
        private var queryC: String? = null
        private var queryD: String? = null

        @JvmField
        val CREATOR: Parcelable.Creator<*> = object : Parcelable.Creator<Any?> {
            override fun createFromParcel(`in`: Parcel): Query {
                return Query(`in`)
            }

            override fun newArray(size: Int): Array<Query?> {
                return arrayOfNulls(size)
            }
        }
    }
}